test/multimodal_model/test_multimodal_model.py

```diff
@@ -71,3 +90,37 @@ def test_llama_mm_vision_encoder(self):
         # assert torch.allclose(
         #     output.mean(), torch.tensor(5.28800), atol=1e-3, rtol=1e-3
         # )
+
+
+class TestMultimodalModelDecoder:
+    @pytest.fixture(autouse=True)
+    def setup_class(self, decoder_config):
+        self.model_args = decoder_config
+        self.batch_size = 1
+        self.dim = self.model_args.dim
+        self.vocab_size = self.model_args.vocab_size
+        self.seq_len = 128
+        self.input = {
+            "tokens": torch.arange(self.batch_size * self.seq_len).reshape(
+                self.batch_size, self.seq_len
+            ),
+            "encoder_input": fixed_init_tensor(
+                (self.batch_size, self.seq_len, self.dim), min_val=-1, max_val=1
+            ),
+            "encoder_mask": None,
+        }
+
+    @torch.no_grad()
+    def test_llama_mm_decoder(self):
+        model = MultimodalDecoder(self.model_args)
+        fixed_init_model(model, min_val=-1, max_val=1)
+
+        # call model
+        output = model(**self.input)
+
+        # assertion
+        expected_shape = (self.batch_size, self.seq_len, self.vocab_size)
+
+        assert (
+            output.shape == expected_shape
+        ), f"Expected shape {expected_shape}, but got {output.shape}"
```

Comment:
Since encoder has numerical testing, shall we also add it for decoder? We can leave them as TODO similar to the encoder part.